Makefile

```diff
@@ -570,24 +625,24 @@ $(TBB_LIB):
 	mkdir -p $(TBB_TMP)
 	mkdir -p $(TBB_TMP_SRC)
 	mkdir -p $(TBB_TMP_BUILD)
-	git clone --branch v2021.4.0 https://github.com/oneapi-src/oneTBB.git $(TBB_TMP_SRC)
+	git clone --branch v2021.7.0 https://github.com/oneapi-src/oneTBB.git $(TBB_TMP_SRC)
 	cd $(TBB_TMP_BUILD)/ && $(CMAKE) $(TBB_TMP_SRC) $(TBB_CMAKEFLAGS)
 	+$(MAKE) -C $(TBB_TMP_BUILD)
 	+$(MAKE) -C $(TBB_TMP_BUILD) install
 	@rm -rf $(TBB_TMP)
 	$(eval undefine TBB_TMP)
 	$(eval undefine TBB_TMP_SRC)
 	$(eval undefine TBB_TMP_BUILD)
+endif
 
 # Eigen
 external_eigen: $(EIGEN_BASE)
 
 $(EIGEN_BASE):
-	# from Eigen master branch as of 2021.08.18
-	git clone -b cms/master/82dd3710dac619448f50331c1d6a35da673f764a https://github.com/cms-externals/eigen-git-mirror.git $@
-	# include all Patatrack updates
-	cd $@ && git reset --hard 6294f3471cc18068079ec6af8ceccebe34b40021
-
+	# from Eigen master branch on gitlab as of 2022.08.09
+	git clone https://gitlab.com/AuroraPerego/eigen.git $@
+	# include all SYCL updates
+	cd $@ && git reset --hard eb029b6bc9c33172f0b77524d226d2bca52945a2
```

Comment:
@fwyzard Would it make sense to clone the `eigen` under `cms-patatrack`? I'd also like to test its impact for other programs separately.

Reply:
When I started, the version of `eigen` in the Makefile wasn't working with SYCL (neither the new version cloned from gitlab).  I took the newest version and put on top of that the changes needed for SYCL (https://gitlab.com/libeigen/eigen/-/merge_requests/1132).
Before this PR I tested serial, native CUDA and  alpaka (serial, CUDA, HIP) with both this version of Eigen and the original one in the Makefile. I've only looked at the throughput and it doesn't change.

Reply:
Regarding this, I did some plots to compare the `eigen` under `cms-patatrack` and this one. 
![Eigen_GPU](https://user-images.githubusercontent.com/84700885/208250955-6fc9d3d4-59c8-40a7-8563-3ecbb85e2e31.png)
![Eigen_CPU](https://user-images.githubusercontent.com/84700885/208250957-f8d282dc-25fc-44aa-a281-d7035df6a088.png)
- new = `eigen` under `gitlab.com/AuroraPerego`
- old  = `eigen` under `cms-patatrack`
They show no loss of performance with the new version